servers/rendering/shader_language.cpp

```diff
@@ -545,63 +545,98 @@ ShaderLanguage::Token ShaderLanguage::_get_token() {
 
 				if (_is_number(GETCHAR(0)) || (GETCHAR(0) == '.' && _is_number(GETCHAR(1)))) {
 					// parse number
+					bool hexa_found = false;
 					bool period_found = false;
 					bool exponent_found = false;
-					bool hexa_found = false;
-					bool sign_found = false;
 					bool float_suffix_found = false;
+					bool uint_suffix_found = false;
+					bool end_suffix_found = false;
+
+					Set<char32_t> symbols;
+					symbols.insert('.');
+					symbols.insert('x');
+					symbols.insert('e');
+					symbols.insert('f');
+					symbols.insert('u');
+					symbols.insert('-');
+					symbols.insert('+');
```

Comment:
Would be good to assess the performance impact of re-constructing `symbols` as a `Set` each loop, as opposed to e.g. storing the valid symbols in constant strings that could be checked against (would likely require storing a bool to know which string to check though, as opposed to mutating `symbols` in the parser).

CC @lawnjelly @bruvzg maybe if you have general insights on this?

Reply:
Okay, as suggested by @lawnjelly I've changed it to using a lut-table (static array of arrays).